data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ScanPartitionCreationSupplier.java

```diff
@@ -101,7 +101,7 @@ private List<PartitionIdentifier> listFilteredS3ObjectsForBucket(final List<Stri
                                                                      final LocalDateTime endDateTime,
                                                                      final Map<String, Object> globalStateMap) {
 
-        Instant mostRecentLastModifiedTimestamp = globalStateMap.containsKey(bucket) ? Instant.parse((String) globalStateMap.get(bucket)) : null;
+        Instant mostRecentLastModifiedTimestamp = globalStateMap.get(bucket) != null ? Instant.parse((String) globalStateMap.get(bucket)) : null;
```

Comment:
This is in another PR. Can you exclude it from this PR in favor of that one?

Reply:
Had to rebase that one